### ISSUE TITLE
Collect the need of the users

### ISSUE THREAD

**Issue by olivier-stasse** [stack-of-tasks/sot-qgv#1]
In this issue, we will collect the users desired features.

0/ Decide in which language the interface will be implemented Python or C++ ? (Python seems to be the best)
1/ Display of the graph at a given frequency.
2/ Creating instances of Nodes
3/ Browsing of the plugins installed on the computer
4/ Plugging Nodes
5/ Interaction with the Nodes commands

**Comment by jmirabel** [comment]
From my understanding, I see two different usage of a GUI:
1. pedagogic usage: one should be able to build a basic graph from scratch, e.g. a simple controller.
2. debugging usage: interesting features are:
  - build the GUI representation of an existing graph,
  - be able to trace a signal value easily,
  - be able to collapse a subset of entity into one. I did develop a GUI for SoT inside gepetto-viewer. In the end, I stopped using it because I did not develop this feature and the graphs I wanted to plot were a way too complex to be visualized correctly.

**Comment by olivier-stasse** [comment]
@jmirabel Thanks for the feedback. Could we start from your Gui ? The interest of a GUI might be seldom when a structure is too complex, but we might find a way to visualize only a subgraph. For instance we could isolate one entity and specify the number of edges we want to display from this entity. Or specify a path for instance.

**Comment by jmirabel** [comment]
It is in https://github.com/jmirabel/sot-gepetto-viewer. I provide it as an example. It may not be the best way of implementing it.

The main challenge is the communication. There are two options:
1. execute the GUI in the same process as SoT,
2. execute the GUI in a separate process (e.g. through *run_command*).

Solution 1 is simple, suited for usage 1 above but cannot be used with the way we plug SoT into ROS). Solution 2 is more difficult to implement and more suited to usage 2 above.
My code implements solution 2 using `run_command` as a mean of communication.

**Comment by olivier-stasse** [comment]
From @tflayols      

I was looking at possible options a while ago, and the best thing for me would be to be able to enter a graph and edit it as well. A live interaction with the graph to view, edit parameters etc. would also be awesome.
https://github.com/paceholder/nodeeditor
https://www.blenderfreak.com/tutorials/node-editor-tutorial-series/
https://pypi.org/project/qtpynodeeditor/

**Comment by olivier-stasse** [comment]
Apart from https://github.com/jmirabel/sot-gepetto-viewer/blob/3a21da95d85f295668157abc3ecccff1c45b9e84/src/sot_gepetto_viewer/sot_plugin.py#L5 there is nothing specific to gepetto_viewer ?

**Comment by olivier-stasse** [comment]
and https://github.com/jmirabel/sot-gepetto-viewer/blob/3a21da95d85f295668157abc3ecccff1c45b9e84/src/sot_gepetto_viewer/sot_plugin.py#L36
This specific part is also displaying the robot state from the control provided by the SoT.

**Comment by jmirabel** [comment]
As of now, the Python bindings of qgv are inside gepetto-viewer. Otherwise, it should be easy to make this widget independent of gepetto-viewer.

**Comment by olivier-stasse** [comment]
@jmirabel I agree with your statement with respect to make it independent from gepetto-viewer. Then does it make sense to make another repository ? Or should we simply start from yours ?
